contracts/InterchainTokenService.sol

```diff
@@ -551,6 +557,9 @@ contract InterchainTokenService is
         uint256 amount,
         bytes calldata metadata
     ) external payable whenNotPaused {
+        if (sourceAddress == address(0)) revert EmptySourceAddress();
```

Comment:
not really possible, and this is also not equivalent to sourceAddress bytes being empty (this would be converted to 20 zero bytes). So we can skip this check

Reply:
make sense, yes, it will be converted to 20 zero bytes. removed the check